controllers/front/unsubscribe.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ *  @author Clerk.io
+ *  @copyright Copyright (c) 2017 Clerk.io
+ *
+ *  @license MIT License
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+require "ClerkAbstractFrontController.php";
+
+class ClerkUnsubscribeModuleFrontController extends ClerkAbstractFrontController
+{
+
+
+    /**
+     * Get response
+     *
+     * @return array
+     */
+    public function getJsonResponse()
+    {
+        header('User-Agent: ClerkExtensionBot Prestashop/v' . _PS_VERSION_ . ' Clerk/v' . Module::getInstanceByName('clerk')->version . ' PHP/v' . phpversion());
+        header('Content-type: application/json;charset=utf-8');
+
+        // If customer is not CLERK_DATASYNC_SYNC_SUBSCRIBERS then return error
+        if (Configuration::get('CLERK_DATASYNC_SYNC_SUBSCRIBERS', $this->getLanguageId(), null, $this->getShopId()) == '0') {
+            http_response_code(403);
+            return array(
+                'success' => false,
+                'message' => 'Subscriber SYNC is disabled, enable it to handle unsubscribers with Clerk.io'
+            );
+        }
+
+        $email = strtolower(trim(pSQL(Tools::getValue('email'))));
+        if (empty($email)) {
+            http_response_code(400);
```

Comment:
Also 422 or 430

Reply:
422 would imply that it was fine, but i think if there is no value given it is just a bad request?
not sure about 430

Reply:
I was thinking 422 because we cannot process the request without the email, but as long as the URL is correct, the AUTH part is technically correct.